source/module_gint/gint.cpp

```diff
@@ -10,6 +10,17 @@
 #ifdef __MKL
 #include <mkl_service.h>
 #endif
+Gint::~Gint()
+{
+	if(pvpR_alloc_flag)
+	{
+		for(int is =0;is<GlobalV::NSPIN;is++)
+		{
+			delete[] pvpR_reduced[is]; 
+		}
+		delete[] pvpR_reduced;
+	}
+}
```

Comment:
Hi @Qianruipku ,
Could you identify from the report where was `pvpR_reduced` allocated? This free operation should correspond to its allocation.

Reply:
Sorry for late reply, because I didn't realize `gint_k` is a derived class of `gint` at first.

Reply:
It is allocated in functions of gint_k. Only if is allocated, the variable "pvpR_alloc_flag" is true.

Reply:
I see. I suggest doing changes on destructor of `Gint_k`.

```suggestion
Gint_k::~Gint_k()
{
    Gint_k::destroy_pvpR();
}
```

Reply:
I think it's better to deallocated in the destructor of `Gint`. `pvpR_reduced` is the member of `Gint`. `Gint` may allocate `pvpR_reduced` in the future.

Reply:
> I think it's better to deallocated in the destructor of `Gint`. 

`Gint_k::destroy_pvpR()` is a implemented function in `Gint_k`. I still think it's better to put the deallocation at the corresponding part of its allocation, which is in `Gint_k`.

> `pvpR_reduced` is the member of `Gint`. 

@wenfei-li It seems that `Gint::cal_gint` is never used, also its member `pvpR_reduced`? Is this appropriate?

>`Gint` can also allocate `pvpR_reduced` in the future.
Then maybe we could deallocated in the destructor of `Gint` then.

Reply:
It is indeed better to put it in the destructor of gint_k, since pvpR is only used in multi-k calculations. pvpR will never be used in gint_gamma, since there is no need for that.

Reply:
> > I think it's better to deallocated in the destructor of `Gint`.
> 
> `Gint_k::destroy_pvpR()` is a implemented function in `Gint_k`. I still think it's better to put the deallocation at the corresponding part of its allocation, which is in `Gint_k`.
> 
> > `pvpR_reduced` is the member of `Gint`.
> 
> @wenfei-li It seems that `Gint::cal_gint` is never used, also its member `pvpR_reduced`? Is this appropriate?
> 
> > `Gint` can also allocate `pvpR_reduced` in the future.
> 
> Then maybe we could deallocated in the destructor of `Gint` then.
ay
Why do you say cal_gint and pvpR are not used? They are used in lots of places ...

Reply:
> Why do you say cal_gint and pvpR are not used? They are used in lots of places ...

I mean in `Gint`, but not its derived class.

Reply:
> > Why do you say cal_gint and pvpR are not used? They are used in lots of places ...
> 
> I mean in `Gint`, but not its derived class.

I see. Indeed the class Gint is not useful in its own right ... only gint_gamma and gint_k are used